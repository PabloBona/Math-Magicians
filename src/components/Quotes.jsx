import { useEffect, useState } from 'react';

const Quotes = () => {
  const [quotes, setQuotes] = useState([]);
  const [isLoading, setIsLoading] = useState(true);
  const [error, setError] = useState(null);

  useEffect(() => {
    const fetchQuotes = async () => {
      try {
        const response = await fetch('https://api.api-ninjas.com/v1/quotes?category=happiness', {
          headers: {
            'Content-Type': 'application/json',
            'X-Api-Key': 'gNic433/7fQRomOXE7AZ8Q==r2WSl2KJTQfj1hCK',
          },
        });
        if (response.ok) {
          const fetchedQuotes = await response.json();
          setQuotes(fetchedQuotes);
        } else {
          setError('Error fetching quotes.');
        }
      } catch (error) {
        setError('Error fetching quotes. Please try again later.');
      }
      setIsLoading(false);
    };

    fetchQuotes();
  }, []);

  let content = null;
  if (isLoading) {
    content = <div className="col-6 d-flex justify-content-center mt-3 bg-primary text-light rounded">Loading...</div>;
  } else if (error) {
    content = <div className="col-6 alert alert-danger">{error}</div>;
  } else {
    content = (
      <div className="row">
        <ul className="col-12 col-md list-group">
          <li className="p-3 rounded-4 d-flex justify-content-center" key={new Date().getTime().toString()}>
            <div className="d-flex justify-content-center align-items-center p-5 g-3 bg-success rounded w-60">
              <div className="text-light bg-success">
                &quot;
                {quotes[0].quote}
                &quot;
                <hr className="" />
                <div className="d-flex justify-content-center mt-3 bg-success">{quotes[0].author}</div>
              </div>
            </div>
          </li>
        </ul>
      </div>
    );
  }

  return <div className="container d-flex justify-content-center mt-5">{content}</div>;
};

export default Quotes;
