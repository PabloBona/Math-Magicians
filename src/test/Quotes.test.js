import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Quotes from '../components/Quotes';

describe('Quotes component unit test', () => {
  test('Quotes Component Renders Properly', () => {
    const tree = renderer.create(<Quotes />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Quotes Component Displays Loading', () => {
    render(<Quotes />);
    expect(screen.getByText('Loading...')).toBeInTheDocument();
  });
});
