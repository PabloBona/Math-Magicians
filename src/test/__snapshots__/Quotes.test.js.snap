// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quotes component unit test Quotes Component Renders Properly 1`] = `
<div
  className="container d-flex justify-content-center mt-5"
>
  <div
    className="col-6 d-flex justify-content-center mt-3 backCarrot text-light rounded"
  >
    Loading...
  </div>
</div>
`;
